swupd/delta.go

```diff
@@ -0,0 +1,160 @@
+package swupd
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/clearlinux/mixer-tools/helpers"
+	"github.com/pkg/errors"
+)
+
+// CreateDeltas creates all delta files between the previous and current
+// version of the supplied manifest, returning a list of files that
+// delta creation failed for.
+func CreateDeltas(manifest *Manifest, statedir string) ([]*File, error) {
```

Comment:
Do we only care about the errors here, or would it be nice to have at least a count of delta files?

Reply:
Count of total deltas or count of successfully created deltas?

Reply:
I imagine the caller might want to print out or log how many deltas were created successfully after calling this step.